src/library/waveform.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * This is the library for creating the waveform. 
+ * It requires the ID of canvas element where the waveform
+ * will be rendered as initialization input. When the waveform
+ * is created, it requires the analyserNode as real-time audio 
+ * input data for drawing the correct waveform.
+ * 
+ * There is no library dependence, this entire waveform is
+ * created based on the canvas element.
+ */
+
+class Waveform {
+  /**
+   * This is the constructor for initializing waveform
+   * @param {String} canvasId  An ID of a canvas element where
+   *     the waveform will be rendered.
+   * @param {AnalyserNode} analyserNode The analysis node
+   *    which connect with the audio context.
+   */
+  constructor(canvasId, analyserNode) {
+    this.canvas_ = document.querySelector(canvasId);
+    this.canvasContext_ = this.canvas_.getContext('2d');
+    this.width_ = this.canvas_.width;
+    this.height_ = this.canvas_.height;
```

Comment:
What if the size of the canvas changes after the construction?

If that's one of constraints, describe in the constructor comment.